src/middlewared/middlewared/plugins/system/product.py

```diff
@@ -74,13 +74,16 @@ async def product_name(self):
         """
         return "TrueNAS"
 
-    @accepts()
+    @accepts(Bool('fullname', default=True))
     @returns(Str('truenas_version'))
-    def version(self):
-        """
-        Returns software version of the system.
+    def version(self, fullname):
```

Comment:
Should we offer `version` and `full_version` methods instead? This returns completely different things depending on a boolean flag

Reply:
Ok, `version` and `version_number`

Reply:
If it's a CUSTOM or nightly image then it won't be the `version_number`, it'll be `23.10-MASTER-timestamp-here` same thing for custom.